src/apns_connection.erl

```diff
@@ -167,100 +180,208 @@ push_notification(ConnectionId, DeviceId, Notification, Headers) ->
                        , notification()
                        , apns:headers()) -> apns:response() | {error, not_connection_owner}.
 push_notification(ConnectionId, Token, DeviceId, Notification, Headers) ->
-  case gen_server:call(ConnectionId, {push_notification, Token, DeviceId, Notification, Headers}) of
-    not_connection_owner -> {error, not_connection_owner};
-    {Timeout, StreamId}  -> wait_response(ConnectionId, Timeout, StreamId)
+  gen_statem:call(ConnectionId, {push_notification, Token, DeviceId, Notification, Headers}).
+
+%% @doc Waits until receive the `connection_up` message
+-spec wait_apns_connection_up(pid()) -> ok | {error, timeout}.
+wait_apns_connection_up(Server) ->
+  receive
+    {connection_up, Server} -> ok;
+    {timeout, Server}       -> {error, timeout}
```

Comment:
I couldn't find the place where this timeout message is sent. I suspect it not being sent at all

Reply:
Can probably just be converted to a gen_statem:call that's only handled by the connected/3 state function.

Reply:
i have a example code :
https://github.com/boboxiaodd/apns/blob/master/apns_worker.erl